iree/turbine/kernel/wave/analysis/index_sequence_analysis.py

```diff
@@ -316,19 +317,23 @@ def populate_read_write_source_indices(
             else node.elements_per_thread
         )
 
-        if elements_per_thread is None:
-            raise ValueError(
-                f"Elements per thread not set for {node} with indexing dim {dim}"
-            )
-
         stride = compute_stride(
             node.indexing_dims, hardware_constraint.vector_shapes, dim
         )
         wg_constraint = [x for x in workgroup_constraints if x.dim == dim]
+
+        assert len(wg_constraint) <= 1, f"Multiple workgroup constraints for dim {dim}"
         if not wg_constraint:
             continue
+
+        wg_dim = wg_constraint[0].workgroup_dim
+        if elements_per_thread is None:
+            tile_size = wg_constraint[0].tile_size
+            threads_count = hardware_constraint.threads_per_block[wg_dim]
```

Comment:
Right now, we allow workgroup_dims > 2, wouldnt that create a problem here when we try to access threads_per_block > 2? Maybe add an assert with a TODO that we should allow threads_per_block > 2 ?

Reply:
Good idea, done